src/chrome/chromeConnection.ts

```diff
@@ -192,4 +191,22 @@ export class ChromeConnection implements IObservableEvents<IStepStartedEventsEmi
                 return (version) ? version : new TargetVersions(Version.unknownVersion(), Version.unknownVersion());
             });
     }
+
+    /**
+     * Gets the port on which chrome was launched, and throw error if the port is not open or accepting connections
+     * @param host The host address on which to check if the port is listening
+     * @param userDataDir Chrome user data directory in which to check for a port file
+     */
+    private async getLaunchedPort(host: string, userDataDir: string): Promise<number> {
+        logger.verbose('Looking for DevToolsActivePort file...');
+        const launchedPort = await chromeUtils.getLaunchedPort(userDataDir);
+        logger.verbose('Got the port, checking if its ready...');
+        const portInUse = await chromeUtils.isPortInUse(launchedPort, host, 100);
```

Comment:
Do we need to verify that  the port is in use? Would it work if we just call this._targetDiscoveryStrategy.getTarget and let it fail and retry if the port is not in use?

Reply:
We could do that, but the reason I chose to test the port explicitly was A) we know that if nothing is listening on that port, we will never be able to succesfully get targets anyway, and B) it seemed difficult to adjust the default timeout specifically for the socket connect portion of the HTTP request (which is 2s by default), this meant that often we were actually waiting 4s  to connect instead of 2.2 or whatever. We could just use promiseTimeout from our utils on that and throw away the result, but that seemed a little messy to me.